src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,334 @@
+import {
+    Component,
+    Directive,
+    Input,
+    Output,
+    EventEmitter,
+    OnInit,
+    OnDestroy,
+    AfterViewInit,
+    ElementRef,
+    Injectable,
+    Optional,
+    Renderer2,
+    ViewContainerRef,} from '@angular/core';
+
+import {Observable} from 'rxjs/Observable';
+import {ScrollDispatcher} from '../core/overlay/scroll/scroll-dispatcher';
+import {Scrollable} from '../core/overlay/scroll/scrollable';
+import {Subject} from 'rxjs/Subject';
+
+
+@Directive({
+    selector: '[md-sticky-viewport], [cdkStickyViewport], [cdkStickyRegion]',
+})
+
+
+@Injectable()
+export class StickyParentDirective implements OnInit, OnDestroy, AfterViewInit {
+
+    // The parent element, which with the 'md-sticky-viewport' tag
+    private pelem: any;
+
+    constructor(private element: ElementRef) {
+        this.pelem = element.nativeElement;
+    }
+
+    ngOnInit(): void {
+        this.pelem.classList.add('sticky-parent');
+    }
+
+    ngAfterViewInit(): void {
+
+    }
+
+    ngOnDestroy(): void {
+        this.pelem.classList.remove('sticky-parent');
+    }
+
+    getElementRef(): ElementRef {
+        return this.element;
+    }
+}
+
+
+
+@Directive({
+    selector: '[md-sticky], [cdkSticky], [cdkStickyHeader]',
+})
+
+
+@Injectable()
+export class StickyHeaderDirective implements OnInit, OnDestroy, AfterViewInit {
+
+    @Input('sticky-zIndex') zIndex: number = 10;
```

Comment:
Added comments for zIndex:
/**Set the sticky-header's z-index as 10 in default. Make it as an input
     * variable to make user be able to customize the zIndex when
     * the sticky-header's zIndex is not the largest in current page.
     * Because if the sticky-header's zIndex is not the largest in current page,
     * it may be sheltered by other element when being sticked.
*/